src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -2636,16 +2639,19 @@ private static void GenerateTypesAndUsings(ScriptBlockAst rootForDefiningTypesAn
                     GenerateLoadUsings(rootForDefiningTypesAndUsings.UsingStatements, allUsingsAreNamespaces, exprs);
                 }
 
-                var typeAstList = new List<TypeDefinitionAst>();
-                foreach (TypeDefinitionAst ast in rootForDefiningTypesAndUsings.FindAll(ast => ast is TypeDefinitionAst, searchNestedScriptBlocks: true))
-                {
-                    typeAstList.Add((TypeDefinitionAst)ast);
-                }
-
-                TypeDefinitionAst[] typeAsts = typeAstList.ToArray();
+                List<Ast> typeAstList = AstSearcher.FindAll(
```

Comment:
Can we:
```suggestion
                List<TypeDefinitionAst> typeAstList = AstSearcher.FindAll(
```
and exclude castings below?

Reply:
🤔 Not sure, to be honest. I guess I can try and see if it throws a fit.

Reply:
We can't do that directly, but if we `.ToArray()` the result we can then cast directly to `TypeDefinitionAst[]`. I'll amend this with that change.

Reply:
Heh, nevermind, that cast breaks too. So nope, no way 'round this at the moment.